source/normalize-arguments.js

```diff
@@ -13,6 +13,47 @@ const knownHookEvents = require('./known-hook-events');
 
 const retryAfterStatusCodes = new Set([413, 429, 503]);
 
+const preNormalize = options => {
```

Comment:
I've split `normalize-argument.js` into two parts:
- `preNormalize` handles all the stuff which is related with static options like `baseUrl`, `followRedirect`, `hooks` etc.
  It is used in `got.create()` to  normalize `defaults.options`.
 - `normalize` does `preNormalize` + handles all the stuff related with dynamic options like `url`, `headers`, `body` etc.